homeassistant/components/crownstone/config_flow.py

```diff
@@ -0,0 +1,229 @@
+                selection = user_input[CONF_USB_PATH]
+
+                # no usb should be used, finish flow
+                if selection == DONT_USE_USB:
+                    return self.async_finish_flow()
+
+                # show a form with text field to enter manual path
+                if selection == MANUAL_PATH:
+                    return await self.async_step_usb_manual_config()
+
+                # user can refresh the list while in this step
+                if selection != REFRESH_LIST:
+                    # get serial-id info
+                    selected_port: ListPortInfo = list_of_ports[
+                        (ports_as_string.index(selection) - 1)
+                    ]
+                    self.usb_path = await self.hass.async_add_executor_job(
+                        get_serial_by_id, selected_port.device
+                    )
+
+                    # check if we are updating an existing entry
+                    if self.existing_entry_id is not None:
+                        return await self.async_update_existing_entry_usb_path()
+
+                    return self.async_finish_flow()
+
+        return self.async_show_form(
+            step_id="usb_config",
+            data_schema=vol.Schema(
+                {vol.Required(CONF_USB_PATH): vol.In(ports_as_string)}
+            ),
+        )
+
+    async def async_step_usb_manual_config(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Manually enter Crownstone USB dongle path."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="usb_manual_config",
+                data_schema=vol.Schema({vol.Required(CONF_USB_MANUAL_PATH): str}),
+            )
+
+        # save usb path
+        self.usb_path = user_input[CONF_USB_MANUAL_PATH]
+
+        # check if we are updating an existing entry
+        if self.existing_entry_id is not None:
+            return await self.async_update_existing_entry_usb_path()
+
+        return self.async_finish_flow()
+
+    async def async_update_existing_entry_usb_path(self) -> FlowResult:
+        """Update usb path of existing entry."""
+        existing_entry = self.hass.config_entries.async_get_entry(
+            str(self.existing_entry_id)
+        )
+        if existing_entry is not None:
+            # copy instead of directly changing memory
+            data = existing_entry.data.copy()
+            data[CONF_USB_PATH] = self.usb_path
+            # update entry & reload
+            self.hass.config_entries.async_update_entry(existing_entry, data=data)
+            await self.hass.config_entries.async_reload(existing_entry.entry_id)
+            # this exits the flow immediately
+            return self.async_abort(reason="usb_setup_complete")
+
+        # if we couldn't find the entry
+        return self.async_abort(reason="usb_setup_unsuccessful")
+
+    def async_finish_flow(self) -> FlowResult:
```

Comment:
Since there are multiple ways a flow can be finished.  Please rename this to show intent.  
```suggestion
    def async_create_new_entry(self) -> FlowResult:
```